packages/material-ui/src/CardMedia/CardMedia.js

```diff
@@ -1,63 +1,89 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { chainPropTypes } from '@material-ui/utils';
-import withStyles from '../styles/withStyles';
+import { chainPropTypes, deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import { getCardMediaUtilityClass } from './cardMediaClasses';
 
-export const styles = {
-  /* Styles applied to the root element. */
-  root: {
-    display: 'block',
-    backgroundSize: 'cover',
-    backgroundRepeat: 'no-repeat',
-    backgroundPosition: 'center',
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+  const { isMediaComponent, isImageComponent } = styleProps;
+
+  return deepmerge(styles.root || {}, {
+    ...(isMediaComponent && styles.media),
+    ...(isImageComponent && styles.img),
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, isMediaComponent, isImageComponent } = styleProps;
+
+  const slots = {
+    root: ['root', isMediaComponent && 'media', isImageComponent && 'img'],
+  };
+
+  return composeClasses(slots, getCardMediaUtilityClass, classes);
+};
+
+const CardMediaRoot = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiCardMedia',
+    slot: 'Root',
+    overridesResolver,
   },
-  /* Styles applied to the root element if `component="video, audio, picture, iframe, or img"`. */
-  media: {
+)(({ styleProps }) => ({
+  /* Styles applied to the root element. */
+  display: 'block',
+  backgroundSize: 'cover',
+  backgroundRepeat: 'no-repeat',
+  backgroundPosition: 'center',
+  ...(styleProps.isMediaComponent && {
+    /* Styles applied to the root element if `component="video, audio, picture, iframe, or img"`. */
     width: '100%',
-  },
-  /* Styles applied to the root element if `component="picture or img"`. */
-  img: {
+  }),
+  ...(styleProps.isImageComponent && {
+    /* Styles applied to the root element if `component="picture or img"`. */
```

Comment:
```suggestion
  /* Styles applied to the root element if `component="picture or img"`. */
  ...(styleProps.isImageComponent && {
```

For consistency with the other components